doc/rbd/rbd-exclusive-locks.rst

```diff
@@ -27,57 +27,55 @@ Exclusive locking is mostly transparent to the user.
+   attempting to write to the image. To disable transparent lock transitions
+   between multiple clients, the client must acquire the lock with
+   ``RBD_LOCK_MODE_EXCLUSIVE``.
 
 
 Blocklisting
 ============
 
-Sometimes, a client process (or, in case of a krbd client, a client
-node's kernel thread) that previously held an exclusive lock on an
-image does not terminate gracefully, but dies abruptly. This may be
-due to having received a ``KILL`` or ``ABRT`` signal, for example, or
-a hard reboot or power failure of the client node. In that case, the
-exclusive lock is never gracefully released. Thus, when a new process
-starts and attempts to use the device, it needs a way to break the
-previously held exclusive lock.
-
-However, a process (or kernel thread) may also hang, or merely lose
-network connectivity to the Ceph cluster for some amount of time. In
-that case, simply breaking the lock would be potentially catastrophic:
-the hung process or connectivity issue may resolve itself, and the old
-process may then compete with one that has started in the interim,
-accessing RBD data in an uncoordinated and destructive manner.
-
-Thus, in the event that a lock cannot be acquired in the standard
-graceful manner, the overtaking process not only breaks the lock, but
-also blocklists the previous lock holder. This is negotiated between
-the new client process and the Ceph Mon: upon receiving the blocklist
-request,
+Sometimes a client process (or, in case of a krbd client, a client node's
+kernel thread) that previously held an exclusive lock on an image does not
+terminate gracefully, but dies abruptly. This may be because the client process
+received a ``KILL`` or ``ABRT`` signal, or because the client node underwent a
+hard reboot or suffered a power failure. In cases like this, the exclusive lock
+is never gracefully released. This means that any new process that starts and
+attempts to use the device must break the previously held exclusive lock.
+
+However, a process (or kernel thread) may hang or merely lose network
+connectivity to the Ceph cluster for some amount of time. In that case,
+breaking the lock would be potentially catastrophic: the hung process or
+connectivity issue could resolve itself and the original process might then
+compete with one that started in the interim, thus accessing RBD data in an
+uncoordinated and destructive manner.
+
+In the event that a lock cannot be acquired in the standard graceful manner,
+the overtaking process not only breaks the lock but also blocklists the
+previous lock holder. This is negotiated between the new client process and the
+Ceph Mon. Upon receiving the blocklist request:
 
 * the Mon instructs the relevant OSDs to no longer serve requests from
   the old client process;
-* once the associated OSD map update is complete, the Mon grants the
+* after the associated OSD map update is complete, the Mon grants the
   lock to the new client;
```

Comment:
This is technically incorrect: the monitor isn't involved with RBD exclusive locking in any way.  It only handles blocklisting, everything else is done by the RBD client.  I'd say: "after the associated OSD map update is complete, the new client can break the previously held lock".

Reply:
The first bullet should probably be amended as well: it's technically correct but I'd move the "upon receiving the blocklist request" bit from the paragraph to the first bullet since it only applies there.

Reply:
Agreed. So altered.